shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/resources/CustomResourceStatus.java

```diff
@@ -0,0 +1,75 @@
+package com.redhat.service.smartevents.shard.operator.v2.resources;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.Optional;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.redhat.service.smartevents.shard.operator.core.resources.Condition;
+import com.redhat.service.smartevents.shard.operator.core.resources.ConditionStatus;
+
+import io.javaoperatorsdk.operator.api.ObservedGenerationAwareStatus;
+
+/**
+ * Common interface for Kubernetes Custom Resource status
+ */
+public abstract class CustomResourceStatus extends ObservedGenerationAwareStatus {
```

Comment:
You could have kept this in `-core` and extended it differently for `-v1` and `-v2`.

Reply:
I moved back `CustomeResourceStatus` to `-core`